monkey/monkey_island/cc/services/bootloader.py

```diff
@@ -0,0 +1,69 @@
+from typing import Dict, List
+
+from bson import ObjectId
+
+from monkey_island.cc.database import mongo
+from monkey_island.cc.services.node import NodeService, NodeNotFoundException
+from monkey_island.cc.services.utils.node_groups import NodeGroups
+from monkey_island.cc.services.utils.bootloader_config import SUPPORTED_WINDOWS_VERSIONS, MIN_GLIBC_VERSION
+
+
+class BootloaderService:
+
+    @staticmethod
+    def parse_bootloader_telem(telem: Dict) -> bool:
+        telem['ips'] = BootloaderService.remove_local_ips(telem['ips'])
+        if telem['os_version'] == "":
+            telem['os_version'] = "Unknown OS"
+
+        telem_id = BootloaderService.get_mongo_id_for_bootloader_telem(telem)
+        mongo.db.bootloader_telems.update({'_id': telem_id}, telem, upsert=True)
+
+        will_monkey_run = BootloaderService.is_os_compatible(telem)
+        try:
+            node = NodeService.get_or_create_node_from_bootloader_telem(telem, will_monkey_run)
+        except NodeNotFoundException:
+            # Didn't find the node, but allow monkey to run anyways
```

Comment:
Maybe log a warning? This is a weird case, no?

Reply:
It happens if monkey is ran on island. Since we don't add island node to the database and generate that node in memory each time, we can't create that node because it would duplicate the island and we can't find it because it does't exist.

Reply:
Renamed it